src/System.Private.CoreLib/shared/System/Threading/ThreadPool.cs

```diff
@@ -685,16 +775,27 @@ private void CleanUp()
             // Since the purpose of calling CleanUp is to transfer any pending workitems into the global
             // queue so that they will be executed by another thread, there's no point in doing this cleanup
             // if we're in the process of shutting down or unloading the AD.  In those cases, the work won't
-            // execute anyway.  And there are subtle races involved there that would lead us to do the wrong
+            // execute anyway.  And there are subtle race conditions involved there that would lead us to do the wrong
             // thing anyway.  So we'll only clean up if this is a "normal" finalization.
-            if (!(Environment.HasShutdownStarted /*|| AppDomain.CurrentDomain.IsFinalizingForUnload()*/))
+            if (!Environment.HasShutdownStarted)
                 CleanUp();
         }
     }
 
     public delegate void WaitCallback(object state);
 
-    public delegate void WaitOrTimerCallback(object state, bool timedOut);  // signalled or timed out
+    public delegate void WaitOrTimerCallback(object state, bool timedOut);  // signaled or timed out
+
+    //
+    // This type is necessary because VS 2010's debugger looks for a method named _ThreadPoolWaitCallbacck.PerformWaitCallback
```

Comment:
This is CoreCLR-specific. (And it is probably not needed anymore.)

Reply:
Ok, will move it out of the shared code.

Reply:
Turns out the comment is quite inaccurate. That's how Dispatch is currently executed from unmanaged code.